base/keylist/keylist.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2024, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+package keylist implements an ordered list (slice) of items,
+with a map from a key (e.g., names) to indexes,
+to support fast lookup by name.
+This is a different implementation of the [ordmap] package,
+that has separate slices for Values and Keys, instead of
+using a tuple list of both. The awkwardness of value access
+through the tuple is the major problem with ordmap.
+*/
+package keylist
+
+import (
+	"fmt"
+	"slices"
+)
+
+// TODO: probably want to consolidate ordmap and keylist
+
+// List implements an ordered list (slice) of Values,
+// with a map from a key (e.g., names) to indexes,
+// to support fast lookup by name.
+type List[K comparable, V any] struct { //types:add
+	// List is the ordered slice of items.
+	Values []V
+
+	Keys []K
```

Comment:
This needs to be exported? And add documentation.

Reply:
yes needs to be exported; docs added.